Source/Core/Util/EncounterSlot.cpp

```diff
@@ -26,7 +26,7 @@ namespace
     template <u32 size, bool greater = false>
     u8 calcSlot(u8 compare, const std::array<u8, size> &ranges)
     {
-        for (size_t i = 0; i < size; i++)
+        for (std::size_t i = 0; i < size; i++)
```

Comment:
I'm a complete C++ noob and I couldn't get the code to compile properly without editing this portion here.
Which I find strange because there are numerous calls to `size_t` elsewhere in the program.

I presume some header file is missing? And I wonder why some code that hasn't been modified in a while is complaining.